### ISSUE TITLE
Error 2 in alignment parsing

### ISSUE THREAD

**Issue by roblanf** [brettc/partitionfinder#74]
Here's another alignment:

```
5 2
a AT
b AT
c AT
d AC

```

The problem here is that the header has one more species than it should. 

Instead of a useful error (which we used to produce), we now get stuck in an infinite loop like this:

```
ERROR    | 2015-10-02 12:42:17,248 | Line 28076, Found no data in file
ERROR    | 2015-10-02 12:42:17,248 | Line 28077, Found no data in file
ERROR    | 2015-10-02 12:42:17,248 | Line 28078, Found no data in file
ERROR    | 2015-10-02 12:42:17,248 | Line 28079, Found no data in file
ERROR    | 2015-10-02 12:42:17,248 | Line 28080, Found no data in file
ERROR    | 2015-10-02 12:42:17,248 | Line 28081, Found no data in file
ERROR    | 2015-10-02 12:42:17,248 | Line 28082, Found no data in file
ERROR    | 2015-10-02 12:42:17,249 | Line 28083, Found no data in file
ERROR    | 2015-10-02 12:42:17,249 | Line 28084, Found no data in file
ERROR    | 2015-10-02 12:42:17,249 | Line 28085, Found no data in file
ERROR    | 2015-10-02 12:42:17,249 | Line 28086, Found no data in file
ERROR    | 2015-10-02 12:42:17,250 | Line 28087, Found no data in file
ERROR    | 2015-10-02 12:42:17,250 | Line 28088, Found no data in file
ERROR    | 2015-10-02 12:42:17,250 | Line 28089, Found no data in file
```

As far as I can tell, this happens because of this:

```
            if num_bits == 0:
                # Skip blanks lines
                continue

```

The issue is that by continuing here, we can stay in the while loop without ever updating the species count. So we NEVER get out of the loop.

@brettc, can you look at this and see if you can fix it?

And can you include this as a test for future-proofing?

R

**Comment by roblanf** [comment]
I thought of just incrementing cur_species before continuing, but I don't think that's sensible here (since the point is that it's a blank line, right?. Thought it was better for you to look at this @brettc, since you know what you want this bit to do.

Maybe better to e.g. check that the number of non-blank lines in the alignment (after the header) is an exact multiple of the number of species? If it's not, then we have a problem with the species count. 

Or maybe we could stream the lines in a way that breaks out of the while loop when you hit EOF?

**Comment by brettc** [comment]
Yep. I now recall making a note to check this all this, and knew I wasn't
finished. But then simply forgot about it.

**Comment by brettc** [comment]
Fixed in 33d919c0ae4dac27c01ed9fc80f51bc08681fae9
